Ci/removing travis

italojs opened the pull request:
Moving travisCI to github actions:

Issues and fixes:

### **issue**: 
`jquery not found`

I dont know how it pass inside travisCI but if you run `./packages/test-in-console/run.sh`or `METEOR_PACKAGE_DIRS='packages/deprecated'   METEOR_NO_DEPRECATION=true TINYTEST_FILTER="logging in with case insensitive email should escape regex special characters" ./meteor test-packages --driver-package test-in-console -p 4096` locally you will se the ...:4096 aplication have an error saying: `jquery not found`
### **Fix**: 
addressed the jquery  inside `test-in-console` app

----------------


### **issue**: 

`EPERM: operation not permitted, chown '/tmp/socket_file_tests'` 


`chownSync` in [main](https://github.com/meteor/meteor/blob/fcb317a31f6e6b9d68a69f6bc9253b60f60d1d70/packages/webapp/webapp_server.js#L1394) fail with `EPERM` on GitHub Actions, because we have a root as default user to handle the files, what isnt allowed in github actions enviroment.

### **Fix**: 

`getWritableGroupName()` picks a real group the current process actually belongs to (via userInfo/getgroups) and uses that for `UNIX_SOCKET_GROUP`, or skips if none is writable.



This PR depends of https://github.com/meteor/meteor/pull/14027 to pass the tests

italojs committed:
Add travis-compat GitHub Actions workflow to run Travis parity tests

- Add .github/workflows/travis-compat.yml
- Trigger on push to ci/removing-travis
- Runs on ubuntu-22.04 with Node 22.17 and C++ toolchain g++-12
- Restores caches (.npm, .meteor, .babel-cache, dev_bundle, chromium), installs system/npm deps and runs packages/test-in-console/run.sh

italojs committed:
Install Puppeteer in test-packages workflow, add jquery to test-in-console, and export METEOR_NO_DEPRECATION earlier in run.sh

italojs committed:
Detect writable group dynamically in webapp socket tests and skip when none found

Add getGroupNameForGid and getWritableGroupName helpers to resolve a writable
group from /etc/group, the current user's gid/groups, or fallbacks (TRAVIS/staff/root).
Use the resolved group for UNIX_SOCKET_GROUP in socket_file_tests and skip the
group-specific test when no writable group can be determined.

imongithubnow committed:
Update app.md

Fix example value for a preference

nachocodoner committed:
Merge pull request #14025 from imongithubnow/patch-1

Update app.md to fix example value for a preference

italojs committed:
Rename workflow job, remove Puppeteer install, add TINYTEST_FILTER, and add email debug logs

- Rename GitHub Actions job travis-compat -> test-packages
- Add TINYTEST_FILTER="email" to test-packages workflow env
- Remove global Puppeteer install step and drop "Travis parity" label from run step
- Add console.log debug output in Email.sendAsync to show MAIL_URL env, settings, and Meteor.isProduction before the production check

italojs committed:
Remove console.log debug output and include MAIL_URL, settings, and production flag in missing-mail error message

italojs committed:
Unquote TINYTEST_FILTER value and add NODE_ENV: CI to test-packages workflow

italojs committed:
Remove MAIL_URL, settings and production flag from missing-mail error message

Previously the thrown error included the MAIL_URL environment value, package
settings JSON, and Meteor.isProduction flag. That could leak sensitive
configuration into logs. Replace it with the original concise guidance and
link to the docs.

italojs committed:
Remove TINYTEST_FILTER from test-packages workflow

italojs committed:
Merge branch 'devel' into ci/removing-travis

italojs commented:
will merge this PR for change streams proposal, but maybe I will move to release 3.4.1

nachocodoner commented:
Yeah, it would be good to cherry pick these changes and put them ready from devel based branches.

Didn't you face any flakiness on the new check? Recently got some issues on using GA runners, but maybe you didn't get any problem with it.

nachocodoner commented on packages/webapp/socket_file_tests.js:
```diff
@@ -131,9 +174,16 @@ testAsyncMulti(
+        // Skip when no writable group could be determined for the current user.
+        test.isTrue(true);
+        return;
+      }
```
Not sure if by doing this we will get some sort of uncover behavior on the new check. Have you seen if as part of the check at least it does what you mentioned in the description:

> getWritableGroupName() picks a real group the current process actually belongs to (via userInfo/getgroups) and uses that for UNIX_SOCKET_GROUP

Or does it skips all time?

italojs committed:
Disable ambiguous error messages in password tests (#14027)

Set Accounts._options.ambiguousErrorMessages = false in password_tests.js
so tests receive specific, deterministic error reasons instead of ambiguous
messages.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/meteor/meteor?pullRequest=14026) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/meteor/meteor?pullRequest=14026) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: italojs<br/>:white_check_mark: nachocodoner<br/>:x: imongithubnow<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/meteor/meteor?pullRequest=14026) it.</sub>
